### PR TITLE
C/cython warnings cleanup

### PR DESCRIPTION
**Pull request by BubuLK**
C/cython code cleanup to remove to many warnings (issue #406).
Partially duplicated work (see PR #423).
Still some additional work need to be done...

### CODE REVIEW

File: sfepy/discrete/common/extmods/mesh.c [hunk 1]
```diff
@@ -891,15 +891,15 @@ int32 mesh_get_centroids(Mesh *mesh, float64 *ccoors, int32 dim)
   MeshEntityIterator it0[1], it1[1];
 
   for (mei_init(it0, mesh, dim); mei_go(it0); mei_next(it0)) {
-    for (id = 0; id < nc; id++) {
+    for (id = 0; id < (uint32)nc; id++) {
```
Comment (rc) [inline sfepy/discrete/common/extmods/mesh.c]: uint32 nc;

File: sfepy/discrete/common/extmods/mesh.c [hunk 2]
```diff
@@ -927,7 +927,7 @@ inline float64 _tri_area(float64 *coors, uint32 *indices, int32 nc)
     v1[2] = 0.0;
   }
 
-  for (id = 0; id < nc; id++) {
+  for (id = 0; id < (uint32)nc; id++) {
```
Comment (rc) [inline sfepy/discrete/common/extmods/mesh.c]: uint32 nc in the function definition

File: sfepy/discrete/common/extmods/refcoors.c [hunk 3]
```diff
@@ -213,7 +213,7 @@ int32 refc_find_ref_coors_convex(FMField *ref_coors,
                        normals0->val + nc * noffs[cell]);
 
         tmin = 1e10;
-        for (ii = 0; ii < loc->num; ii++) {
+        for (ii = 0; ii < (int32)loc->num; ii++) {
```
Comment (rc) [inline sfepy/discrete/common/extmods/refcoors.c]: uint32 ii, ...

File: sfepy/discrete/iga/extmods/igac.pyx [hunk 4]
```diff
@@ -351,7 +351,7 @@ def eval_mapping_data_in_qp(np.ndarray[float64, mode='c', ndim=2] qps not None,
     dx_dxi = np.empty((1, 1, dim, dim), dtype=np.float64)
     dxi_dx = np.empty((1, 1, dim, dim), dtype=np.float64)
 
-    for ii in range(0, dim):
+    for ii in range(0, <uint32>dim):
```
Comment (rc) [inline sfepy/discrete/iga/extmods/igac.pyx]: Did this generate warnings too?

**Comment by rc** [comment]
Thanks! I have just found a few more cases where retyping the variables instead of retyping a loop bound would be better.

**Comment by rc** [comment]
Otherwise all looks good - the undefined vars are solved by #423.

File: sfepy/discrete/common/extmods/mesh.c [hunk 5]
```diff
@@ -107,8 +107,8 @@ int32 mesh_print(Mesh *mesh, FILE *file, int32 header_only)
     }
 
     fprintf(file, "topology connectivities:\n");
-    for (ii = 0; ii <= D; ii++) {
-      for (id = 0; id <= D; id++) {
+    for (ii = 0; ii <= (uint32)D; ii++) {
+      for (id = 0; id <= (uint32)D; id++) {
```
[the remark this answers is not in the document]
Comment (BubuLK) [reply sfepy/discrete/common/extmods/mesh.c]: Already done (uint32 ii,...) but...
int32 D = mesh->topology->max_dim - should be THIS declaration changed? Probably not (?)

File: sfepy/discrete/common/extmods/mesh.c (hunk 1 above)
Comment (BubuLK) [reply sfepy/discrete/common/extmods/mesh.c]: nc - same as above comment...

File: sfepy/discrete/common/extmods/mesh.c (hunk 2 above)
Comment (BubuLK) [reply sfepy/discrete/common/extmods/mesh.c]: Dtto...

File: sfepy/discrete/common/extmods/refcoors.c (hunk 3 above)
Comment (BubuLK) [reply sfepy/discrete/common/extmods/refcoors.c]: Done. Same as above for nc (line 367).

File: sfepy/discrete/iga/extmods/igac.pyx (hunk 4 above)
Comment (BubuLK) [reply sfepy/discrete/iga/extmods/igac.pyx]: Yes... not from cython, but following generated c-code compilation.

File: sfepy/discrete/common/extmods/mesh.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (rc) [reply sfepy/discrete/common/extmods/mesh.c]: Yes! `mesh->topology->max_dim` is `uint32`, see `mesh.h`.

File: sfepy/discrete/common/extmods/mesh.c (hunk 1 above)
Comment (rc) [reply sfepy/discrete/common/extmods/mesh.c]: Also yes :) `mesh->geometry->dim` is uint32

File: sfepy/discrete/common/extmods/mesh.c (hunk 2 above)
Comment (rc) [reply sfepy/discrete/common/extmods/mesh.c]: Yes, basically, all `D`, `dim`, `nc` can/should be changed to uint32.

File: sfepy/discrete/iga/extmods/igac.pyx (hunk 4 above)
Comment (rc) [reply sfepy/discrete/iga/extmods/igac.pyx]: OK, thanks!

**Comment by rc** [comment]
LGTM, thanks!

**Pull request merged**